### PR TITLE
Layer Styling Components

### PR DESCRIPTION
**Pull request by glenselle**

### CODE REVIEW

File: src/LayerStyler/StyleManager/index.js [hunk 1]
```diff
@@ -0,0 +1,325 @@
...
+    'olKit.LabelStyler.color': 'Color',
+    'olKit.LabelStyler.outline': 'Outline',
+    'olKit.LabelStyler.textHeight': 'Text Height',
+    'olKit.LabelStyler.textUnits': 'Text Size Units',
+    'olKit.LabelStyler.pixels': 'Pixels',
+    'olKit.LabelStyler.feet': 'Feet',
+    'olKit.LabelStyler.chooseAttrs': 'Choose Attributes to Display',
+    'olKit.LabelStyler.clear': 'Clear All'
+  }
+}
```
Comment (PatchesMaps) [inline src/LayerStyler/StyleManager/index.js:323]: move to locales

File: src/LayerStyler/_ColorPicker/index.js [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+  '#ff9800',
+  '#795548',
+  '#607d8b',
+  '#000000',
+  '#ffffff'
+]
+
+/**
+ * @component
+ * @category vmc
```
Comment (PatchesMaps) [inline src/LayerStyler/_ColorPicker/index.js:33]: rip out vmc

File: src/LayerStyler/LayerStyler.js [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+        onDefaultStyleChange={this.onDefaultStyleChange}
+        onDefaultStyleReset={this.onDefaultStyleReset}
+        {...this.props} />
+    )
+  }
+}
+
+LayerStyler.propTypes = {
+  /** Object with key/value pairs for translated strings */
+  translations: PropTypes.object.isRequired,
```
Comment (PatchesMaps) [inline src/LayerStyler/LayerStyler.js:203]: translations is required

Comment (PatchesMaps) [reply src/LayerStyler/LayerStyler.js:203]: connectToMap instead of requiring translations

File: src/LayerStyler/LayerStyler.js [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+
+  /** Openlayers map object */
+  map: PropTypes.object.isRequired
+}
+
+export default LayerStyler
```
Comment (PatchesMaps) [inline src/LayerStyler/LayerStyler.js:209]: connectToMap()

File: src/LayerStyler/_SelectTabs/index.js [hunk 5]
```diff
@@ -0,0 +1,73 @@
+import React, { Component } from 'react'
+import PropTypes from 'prop-types'
+
+import { Flex, TabButton, TabList, TabsContainer, TabSlider } from './styled'
+
+/**
+ * @component
+ * @category SelectTabs
+ */
```
Comment (PatchesMaps) [inline src/LayerStyler/_SelectTabs/index.js:9]: remove these on un-exported comps

File: src/classes/VectorLayer.js [hunk 6]
```diff
@@ -82,7 +82,7 @@ class VectorLayer extends olLayerVector {
    * @since NEXT
    */
   setDefaultVectorStyles () {
-    return this.parser.readStyle(this.getStyle()()).then(style => {
+    this.parser.readStyle(this.getStyle()()).then(style => {
```
Comment (PatchesMaps) [inline src/classes/VectorLayer.js:85]: add return back in

File: src/classes/VectorLayer.js [hunk 7]
```diff
@@ -25,7 +25,7 @@ class VectorLayer extends olLayerVector {
     this.defaultStyles = []
     this._defaultStylesCache = []
     this.isVectorLayer = true
-    this._setInitialStyle()
+    if (!opts.style) this._setInitialStyle()
```
Comment (PatchesMaps) [inline src/classes/VectorLayer.js:28]: revert this

File: babel.config.js [hunk 8]
```diff
@@ -6,6 +6,7 @@ module.exports = {
   'plugins': [
     ['@babel/plugin-proposal-class-properties'],
     ['@babel/plugin-transform-runtime'],
+    ['@babel/plugin-proposal-optional-chaining'],
```
Comment (stazrad) [inline babel.config.js:9]: yessss 🙌

File: package.json [hunk 9]
```diff
@@ -53,26 +53,35 @@
     "url": "git@github.com:MonsantoCo/ol-kit.git"
   },
   "dependencies": {
+    "@material-ui/core": "^4.9.11",
+    "@material-ui/icons": "^4.9.1",
```
Comment (stazrad) [inline package.json:57]: material ui deps need to be `peerDependency` & `devDependency`

File: package.json [hunk 10]
```diff
@@ -53,26 +53,35 @@
...
     "@turf/bbox": "^6.0.1",
     "@turf/centroid": "^6.0.2",
     "@turf/helpers": "^6.1.4",
     "geostyler-openlayers-parser": "0.21.0",
+    "lodash.clonedeep": "^4.5.0",
     "lodash.debounce": "~4.0.8",
-    "nanoid": "~2.1.6",
+    "lodash.escaperegexp": "^4.1.2",
+    "material-ui": "^0.20.2",
```
Comment (stazrad) [inline package.json:65]: is this one also needed?

Comment (PatchesMaps) [reply package.json:65]: hmmm, I honestly don't know what that is but I don't think it should be there

File: src/LayerStyler/StyleManager/index.js [hunk 11]
```diff
@@ -0,0 +1,283 @@
...
+  /** returns a string which is the title of the layer given an OL layer */
+  getTitleForLayer: PropTypes.func.isRequired,
+
+  /** returns an array of strings representing the values for the selected attribute */
+  getValuesForAttribute: PropTypes.func.isRequired,
+
+  /** array of strings representing values for the selected attribute */
+  attributeValues: PropTypes.array.isRequired,
+
+  /** @since 6.6.0
```
Comment (stazrad) [inline src/LayerStyler/StyleManager/index.js:266]: drop this `@since`

Comment (PatchesMaps) [reply src/LayerStyler/StyleManager/index.js:266]: good catch

File: src/Map/utils.js [hunk 12]
```diff
@@ -62,7 +62,7 @@ export function connectToMap (Component) {
       ? <Component {...props} />
       : (
         <MapContext.Consumer>
-          {providerProps => {
+          {(providerProps = {}) => {
```
Comment (stazrad) [inline src/Map/utils.js:65]: 👍

File: src/LayerStyler/utils.js [hunk 13]
```diff
@@ -0,0 +1,25 @@
...
+  const slowDebounce = debounce(callback, 0)
+  const fastDebounce = debounce(callback, 0)
+  const keys = [
+    map.on('change:size', slowDebounce, thisObj),
+    map.getView().on('change:resolution', slowDebounce, thisObj),
+    map.getView().on('change:center', fastDebounce, thisObj)
+  ]
+
+  return keys
+}
```
Comment (stazrad) [inline src/LayerStyler/utils.js:25]: this whole util exists in the popup utils & this export will collide with that one....finally a case to hoist a common util??

**Pull request merged**